### PR TITLE
Actions to preview docs

### PR DESCRIPTION
**Pull request by luanbrdev**
Hey guys, how are you? 

@gastaldi, @ricardozanini

Like we talk in conversation PR https://github.com/quarkiverse/quarkus-openapi-generator/pull/493, I'm trying to add the actions to run the preview. But I don't know like test and I'm not sure if it's the correct way.

### CODE REVIEW

File: .github/workflows/preview.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+          workflow_conclusion: success
+          name: site
+      - name: Store PR id as variable
+        id: pr
+        run: |
+          echo "id=$(<pr-id.txt)" >> $GITHUB_OUTPUT 
+          rm -f pr-id.txt
+      - name: Publishing to surge for preview
+        id: deploy
+        run: npx surge ./ --domain https://quarkus-website-pr-${{  steps.pr.outputs.id }}-preview.surge.sh --token ${{ secrets.SURGE_TOKEN }}
```
Comment (ricardozanini) [inline .github/workflows/preview.yml:27]: @gastaldi shouldn't we add a prefix here?

**Review by ricardozanini — COMMENTED** [review]
I think the only thing we need to change is the website URL to not clash with Quarkus main.

File: .github/workflows/build-adoc.yml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    strategy:
+     matrix:
+        os: [ubuntu-latest, windows-latest]
+        java: [
+          { 'version': '11' },
+          { 'version': '17' }
+        ]
+    steps:
+      - name: Pre preview
+        run: echo Pre preview
```
Comment (gastaldi) [inline .github/workflows/build-adoc.yml:21]: This doesn't do anything. You need to build the docs module and generate the website content.

File: .github/workflows/preview.yml (hunk 1 above)
Comment (luanbrdev) [reply .github/workflows/preview.yml]: Hi @ricardozanini, I changed the url to https://quarkus-openapi-generator-preview-pr-...

File: .github/workflows/build-adoc.yml (hunk 2 above)
Comment (luanbrdev) [reply .github/workflows/build-adoc.yml]: Hi @gastaldi, I changed the file and added the jekyll, I added also gemfile and _config.yml. Let me know if was a mistake and any missing steps.

**Comment by gastaldi** [comment]
I think we need to get to a common sense to have this feature for all Quarkiverse extensions. I have opened https://github.com/quarkusio/quarkus/issues/36289 to investigate this.

File: .github/workflows/preview.yml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+      - name: Store PR id as variable
+        id: pr
+        run: |
+          echo "id=$(<pr-id.txt)" >> $GITHUB_OUTPUT 
+          rm -f pr-id.txt
+      - name: Publishing to surge for preview
+        id: deploy
+        run: npx surge ./ --domain https://quarkus-openapi-generator-preview-pr-${{  steps.pr.outputs.id }}.surge.sh --token ${{ secrets.SURGE_TOKEN }}
+      - name: Update PR status comment on success
+        uses: actions-cool/maintain-one-comment@v3.0.0
```
Comment (ricardozanini) [inline .github/workflows/preview.yml:29]:
```suggestion
        uses: actions-cool/maintain-one-comment@v3.1.1
```

File: .github/workflows/preview.yml [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+          token: ${{ secrets.GITHUB_TOKEN }}
+          body: |
+            🎊 PR Preview ${{ github.sha }} has been successfully built and deployed to https://quarkus-openapi-generator-preview-pr-${{ steps.pr.outputs.id }}.surge.sh
+            <img width="300" src="https://user-images.githubusercontent.com/507615/90250366-88233900-de6e-11ea-95a5-84f0762ffd39.png">
+            <!-- Sticky Pull Request Comment -->
+          body-include: '<!-- Sticky Pull Request Comment -->'
+          number: ${{ steps.pr.outputs.id }}
+      - name: Update PR status comment on failure
+        if: ${{ failure() }}
+        uses: actions-cool/maintain-one-comment@v3.0.0
```
Comment (ricardozanini) [inline .github/workflows/preview.yml:40]:
```suggestion
        uses: actions-cool/maintain-one-comment@v3.1.1
```

File: .github/workflows/preview_teardown.yml [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+jobs:
+  preview-teardown:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Teardown surge preview
+        id: deploy
+        run: npx surge teardown https://quarkus-openapi-generator-preview-pr-${{ github.event.number }}.surge.sh --token ${{ secrets.SURGE_TOKEN }}
+      - name: Update PR status comment
+        uses: actions-cool/maintain-one-comment@v3.0.0
```
Comment (ricardozanini) [inline .github/workflows/preview_teardown.yml:17]:
```suggestion
        uses: actions-cool/maintain-one-comment@v3.1.1
```

File: .github/workflows/preview.yml [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+      - name: Download PR Artifact
+        uses: dawidd6/action-download-artifact@v2
+        with:
+          workflow: ${{ github.event.workflow_run.workflow_id }}
+          workflow_conclusion: success
+          name: site
+      - name: Store PR id as variable
+        id: pr
+        run: |
+          echo "id=$(<pr-id.txt)" >> $GITHUB_OUTPUT
```
Comment (ricardozanini) [inline .github/workflows/preview.yml:23]: We need this file generated either in build or Build Adoc. 

@gastaldi I guess we can save this file in Build Adoc with:

```
    - name: Store PR id
      run: |
        echo ${{ github.event.number }} > pr-id.txt
```

This workflow is slightly different from the one you posted here: 

https://github.com/quarkiverse/quarkus-prettytime/blob/main/.github/workflows/build.yml#L43C8-L52

@luanbrdev Can you try running locally your PR and see if we get the `./docs/target/generated-docs` dir?

Then, I think we can use the same approach and have the website built in the maven build process instead. Here's the reference: https://github.com/quarkiverse/quarkus-prettytime/tree/main/.github/workflows

Comment (luanbrdev) [reply .github/workflows/preview.yml:23]: Hi, what's up?! I will use the reference. But, I have a question, in the build.yml file in our project we have a path-ignore for adoc files (if I'm understand correctly). When I created build-adoc.yml I was think about only run the preview in cases of adoc files update. This make sense? 
About run the actions locally, I dunno how to do this, but I will search about. :stuck_out_tongue_winking_eye:

Comment (ricardozanini) [reply .github/workflows/preview.yml:23]: > When I created build-adoc.yml I was think about only run the preview in cases of adoc files update. This make sense?

Yes, indeed! You're right, we can keep both workflows separate. 

> About run the actions locally, I dunno how to do this, but I will search about

You can mimic what the workflow is doing. 👍

**Comment by ricardozanini** [comment]
Sounds nice! Have you run it locally and were able to generate the website?

**Comment by luanbrdev** [comment]
Hey @ricardozanini, how are you?
Well, when I run the project with mvn the index.html is generated correctly in /docs/target/generated-docs/. 

About the actions, I could not run completely, what I did was, I executed the job "build" using (nektos/act) and added one step before _Publishing directory for PR preview_ that printed all files in directory /docs/target/generated-docs/ and the index.html was generated. I think that it's working, but I could not run completely. 

About jobs, preview and preview_teardown I only followed the example, but I not executed them.

**Review by ricardozanini — APPROVED** [review]
@gastaldi @hbelmiro wanna take a look, please?

File: .github/workflows/build.yml [hunk 7]
```diff
@@ -46,9 +45,25 @@ jobs:
           java-version: ${{matrix.java.version}}
           cache: 'maven'
 
+      - name: Set up Maven
+        uses: stCarolas/setup-maven@v4.5
+        with:
+          maven-version: 3.8.6
```
Comment (gastaldi) [inline .github/workflows/build.yml:51]: Why this is necessary?

Comment (luanbrdev) [reply .github/workflows/build.yml:51]: Hi @gastaldi, what's up?
Well, when I executed locally with (nektos/act) I needed to set this step. If no configuration is necessary to run mvn here, I can remove.

Comment (gastaldi) [reply .github/workflows/build.yml:51]: The `actions/setup-java` already setups Maven, therefore I don't think that's necessary. The only reason I can think of to keep it is if we want to fix the Maven version  but then using the Maven Wrapper (mvnw) is better in this case.

**Comment by luanbrdev** [comment]
Hi guys, 
I understand. I already removed the unnecessary code.

**Review by hbelmiro — APPROVED** [review]

**Comment by luanbrdev** [comment]
Hi @ricardozanini, @gastaldi, @hbelmiro. How are you guys? 
I would like to know if are there more adjusts to do? Let me know.

**Comment by ricardozanini** [comment]
I'm merging this now, then you can rebase your other PR to see if we can get this previews going on!

**Pull request merged**